src/Agoda.Analyzers/AgodaCustom/AG0039MethodLineLengthAnalyzer.cs

```diff
@@ -31,37 +40,29 @@ public class AG0039UndocumentedMemberAnalyzer : DiagnosticAnalyzer
             Category,
             DiagnosticSeverity.Warning,
             isEnabledByDefault: true,
-            description: "Public surfaces should be documented, please add XML documentation.");
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+            helpLinkUri: "https://github.com/agoda-com/AgodaAnalyzers/blob/master/src/Agoda.Analyzers/RuleContent/AG0039UndocumentedMemberAnalyzer.html");
+        
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Descriptor); } }
 
         public override void Initialize(AnalysisContext context)
         {
-            context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType, SymbolKind.Method, SymbolKind.Property, SymbolKind.Field, SymbolKind.Event);
+            context.RegisterSyntaxNodeAction(AnalyzeMethod, SyntaxKind.MethodDeclaration);
         }
 
-        private static void AnalyzeSymbol(SymbolAnalysisContext context)
+        private static void AnalyzeMethod(SyntaxNodeAnalysisContext context)
         {
-            if (context.Symbol.DeclaredAccessibility != Accessibility.Public)
-                return;
-
-            if (!string.IsNullOrWhiteSpace(context.Symbol.GetDocumentationCommentXml()))
-                return;
-            
-            // ignore getters and setter methods for documentation 
-            if (context.Symbol.Kind == SymbolKind.Method)
-            {
-                var methodSymbol = (IMethodSymbol)context.Symbol;
-                if (methodSymbol.MethodKind == MethodKind.PropertyGet ||
-                    methodSymbol.MethodKind == MethodKind.PropertySet)
-                    return;
-            }
+            var methodNode = (MethodDeclarationSyntax)context.Node;
+            var lineCount = GetNonEmptyLineCount(methodNode.GetText().Lines);
 
-            var location = context.Symbol.Locations[0];
-            var name = context.Symbol.Name;
+            if (lineCount <= MaxLines) return;
 
-            var diagnostic = Diagnostic.Create(Descriptor, location, context.Symbol.Kind.ToString(), name);
+            var diagnostic = Diagnostic.Create(Descriptor, methodNode.Identifier.GetLocation(), methodNode.Identifier.Text, lineCount, MaxLines);
             context.ReportDiagnostic(diagnostic);
         }
+
+        private static int GetNonEmptyLineCount(IEnumerable<TextLine> lines)
+        {
+            return lines.Count(line => !string.IsNullOrWhiteSpace(line.ToString()));
```

Comment:
Why do we allow empty lines, but not:
 - comments,
 - lines that only have braces/brackets/etc
 - ...

Feels like the rule in this form will punish certain formatting styles more than others (eg: single line `if` statements).

Reply:
comments i think we should, braces is a think an interesting one if we should include, it may encourage people to use those statements when its better not to you are correct, then again, it may encourage people to use less if and more linq style. its an interesting discussion, i think the best direction is to try and see the impact on the code base, and get feedback from the engineers, i'll make sure to look out for this and also have a chat on slack about how we are measuring the effect. I've got some ideas about using code coach data to track it